### PR TITLE
Update stub for csv

### LINKED ISSUE #1277: Can't use csv.writer with --strict?
When I use `mypy --strict` with a `csv.writer(text_io)`, I get the error `Call to untyped function "writer" in typed context`. I expected that since `csv.writer` is part of the standard library, it would be typed in typeshed, but either it's not typed or I'm misreading:

``` py
def writer(csvfile, dialect=..., **fmtparams): ...
```

Is it possible to work around this error?

### PR DESCRIPTION
**Pull request by ymyzk**
Update stub for csv. Fixes #1277 and #1381.

There are some imprecise type annotations, because I do not now how.

### CODE REVIEW

**Comment by matthiaskramm** [comment]
Would it make sense to merge the Python 2 and Python 3 version of `csv` and `_csv`, while at it?

File: stdlib/3/_csv.pyi [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+class Dialect:
+    delimiter = ...  # type: str
+    quotechar = ...  # type: Optional[str]
+    escapechar = ...  # type: Optional[str]
+    doublequote = ...  # type: bool
+    skipinitialspace = ...  # type: bool
+    lineterminator = ...  # type: str
+    quoting = ...  # type: int
+    def __init__(self) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/_csv.pyi]: At least in master, there's also `strict: int`.

File: stdlib/3/csv.pyi [hunk 2]
```diff
@@ -1,73 +1,74 @@
-from typing import Any, Iterable
+from collections import OrderedDict
+import sys
+from typing import Any, Dict, Iterable, Iterator, List, Optional, Sequence
 
-QUOTE_ALL = ...  # type: int
-QUOTE_MINIMAL = ...  # type: int
-QUOTE_NONE = ...  # type: int
-QUOTE_NONNUMERIC = ...  # type: int
+from _csv import _reader, _writer, reader, writer, register_dialect, unregister_dialect, get_dialect, list_dialects, field_size_limit, QUOTE_ALL, QUOTE_MINIMAL, QUOTE_NONE, QUOTE_NONNUMERIC
```
Comment (JelleZijlstra) [inline stdlib/3/csv.pyi]: To re-export these here, you need to write `from _csv import reader as reader` and similar.

File: stdlib/3/csv.pyi [hunk 3]
```diff
@@ -1,73 +1,74 @@
...
+if sys.version_info >= (3, 6):
+    class DictReader(Iterator[OrderedDict[str, str]]):
+        restkey = ...  # type: Optional[str]
+        restval = ...  # type: Optional[str]
+        reader = ...  # type: _reader
+        # TODO: precise type
+        dialect = ...  # type: Any
+        line_num = ...  # type: int
+        fieldnames = ...  # type: Sequence[str]
+        def __init__(self, f: Iterator[str], fieldnames: Sequence[str] = ..., restkey: Optional[str] =..., restval: Optional[str] = ..., dialect: Any =..., *args: Any, **kwds: Any) -> None: ...
```
Comment (JelleZijlstra) [inline stdlib/3/csv.pyi]: space after each = (also on line 57 below)

**Comment by ymyzk** [comment]
Fixed errors pointed by @JelleZijlstra, and merged the Python 2 version and the Python 3 version.

**Pull request merged**